contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +397,94 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   // --------------------------------------------------------------------------
+   struct blacklist_manager {
+      blacklist_manager(std::vector<name>& blacklist) : _blacklist(blacklist) {}
+
+      void add(const std::vector<name>& patterns) const {
+         std::unordered_set<name, _hash> already;
+
+         for (auto n : _blacklist)
+            already.insert(n);
```

Comment:
Could use `already.insert(_blacklist.begin(), _blacklist.end());` Also in `remove`.

Reply:
Yes, this is nicer, especially since it will `reserve()` the sneeded size. https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/6cb4243178fbeaac7c49986eeb14721fc9347acc